modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/provider/test/APIApplicationProviderTest.java

```diff
@@ -162,6 +165,46 @@ public void test() throws Exception {
 
 		Assert.assertEquals(
 			"name ne 'testName'", filter.getODataFilterString());
+
+		String baseURL = RandomTestUtil.randomString();
+
+		HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"applicationStatus", "published"
+			).put(
+				"baseURL", baseURL
+			).put(
+				"title", "test"
+			).toString(),
+			"headless-builder/applications/" +
+				apiApplicationJSONObject.getLong("id"),
+			Http.Method.PUT);
+
+		apiApplication = _apiApplicationProvider.fetchAPIApplication(
```

Comment:
same situation as in the comment below about maybe using the getPublishedAPIApplications instead